src/main/java/org/spongepowered/api/event/message/MessageEvent.java

```diff
@@ -24,37 +24,254 @@
  */
 package org.spongepowered.api.event.message;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spongepowered.api.text.TextTemplate.arg;
+import static org.spongepowered.api.text.TextTemplate.of;
+
 import org.spongepowered.api.event.Event;
+import org.spongepowered.api.event.impl.AbstractMessageEvent;
+import org.spongepowered.api.eventgencore.annotation.ImplementedBy;
+import org.spongepowered.api.eventgencore.annotation.PropertySettings;
 import org.spongepowered.api.text.Text;
-
-import java.util.Optional;
-
-import javax.annotation.Nullable;
+import org.spongepowered.api.text.TextRepresentable;
+import org.spongepowered.api.text.transform.FixedPartitionedTextFormatter;
+import org.spongepowered.api.text.transform.SimpleTextFormatter;
+import org.spongepowered.api.text.transform.SimpleTextTemplateApplier;
+import org.spongepowered.api.text.transform.TextTemplateApplier;
 
 /**
  * Describes events when a involving a {@link Text} message.
  */
+@ImplementedBy(AbstractMessageEvent.class)
 public interface MessageEvent extends Event {
 
     /**
+     * Parameter for header section in header partition.
+     */
+    String PARAM_MESSAGE_HEADER = "header";
```

Comment:
I personally agree with @kashike. Header/footer sounds like these are additional lines or something, not simply a prefix/suffix.